lib/iris/analysis/cartography.py

```diff
@@ -239,6 +239,24 @@ def _xy_range(cube, mode=None):
     return (x_range, y_range)
 
 
+def _meshgrid(x, y):
```

Comment:
Agreed.  Is this not behaviour that would be useful to have consistent across the entire package?

Perhaps if util were made a sub-package, then it would no longer be such a difficult call to add add additional 'utilities' for fear of further discontinuity:

```Python
iris.util.ndarray.meshgrid
```

In ANTS we have the following structure, which has served us very well:
```Python
ants.utils.ndarray
ants.utils.coord
ants.utils.cube
```